assets/test/view_test.js

```diff
@@ -2,43 +2,188 @@ import LiveSocket, { View } from '../js/phoenix_live_view';
+    let channelStub = {
+      push(evt, payload, timeout) {
+        expect(payload.value).toBe('increment=1')
+        return {
+          receive(status, cb) {
+            let diff = {
+              static: ['<h2>', '</h2>'],
+              fingerprint: 123
+            }
+            cb(diff)
+          }
+        }
+      }
+    }
+    view.channel = channelStub
+
+    view.pushWithReply({ target: el.querySelector('form') }, { value: 'increment=1' })
+
+    expect(view.el.firstChild.tagName).toBe('H2')
+    expect(view.newChildrenAdded).toBe(false)
+  })
+
+  test('pushEvent', function() {
+    expect.assertions(3);
+
+    let liveSocket = new LiveSocket('/live')
+    let el = liveViewDOM()
+    let input = el.querySelector('input')
+
+    let view = new View(el, liveSocket)
+    let channelStub = {
+      push(evt, payload, timeout) {
+        expect(payload.type).toBe('keyup')
+        expect(payload.event).toBeDefined()
+        expect(payload.value).toBe('1')
+        return {
+          receive() {}
+        }
+      }
+    }
+    view.channel = channelStub
+
+    view.pushEvent('keyup', input, { target: input })
+  })
+
+  test('pushKey', function() {
+    expect.assertions(3);
+
+    let liveSocket = new LiveSocket('/live')
+    let el = liveViewDOM()
+    let input = el.querySelector('input')
+
+    let view = new View(el, liveSocket)
+    let channelStub = {
+      push(evt, payload, timeout) {
+        expect(payload.type).toBe('keydown')
+        expect(payload.event).toBeDefined()
+        expect(payload.value).toBe('1')
+        return {
+          receive() {}
+        }
+      }
+    }
+    view.channel = channelStub
+
+    view.pushKey(input, 'keydown', { target: input }, () => {})
+  })
+
+  test('pushInput', function() {
+    expect.assertions(3);
+
+    let liveSocket = new LiveSocket('/live')
+    let el = liveViewDOM()
+    let input = el.querySelector('input')
+
+    let view = new View(el, liveSocket)
+    let channelStub = {
+      push(evt, payload, timeout) {
+        expect(payload.type).toBe('form')
+        expect(payload.event).toBeDefined()
+        expect(payload.value).toBe('increment=1')
+        return {
+          receive() {}
+        }
+      }
+    }
+    view.channel = channelStub
+
+    view.pushInput(input, { target: input })
+  })
+
+  test('submitForm', function() {
+    expect.assertions(7);
+
+    let liveSocket = new LiveSocket('/live')
+    let el = liveViewDOM()
+    let form = el.querySelector('form')
+
+    let view = new View(el, liveSocket)
+    let channelStub = {
+      push(evt, payload, timeout) {
+        expect(payload.type).toBe('form')
+        expect(payload.event).toBeDefined()
+        expect(payload.value).toBe('increment=1')
+        return {
+          receive() {}
+        }
+      }
+    }
+    view.channel = channelStub
+
+    view.submitForm(form, { target: form })
+    expect(form.dataset.phxHasSubmitted).toBeTruthy()
+    expect(form.classList[0]).toBe('phx-loading')
```

Comment:
```suggestion
    expect(form.classList.contains('phx-loading').toBeTruthy()
```

I think this will be more resilient as otherwise the test is assuming `phx-loading` is the first class in the list.